### PR TITLE
Patch for JDK-9

### PR DESCRIPTION
**Pull request by MartyIX**
Hi @epickrram,

This is my patch for JDK-9. There are two things I don't know how to solve:

1) https://github.com/OpenHFT/Chronicle-Bytes/compare/master...MartyIX:jdk9-patch?expand=1#diff-600376dffeb79835ede4a0b285078036R112 - this should be in the profile I guess

2) https://github.com/OpenHFT/Chronicle-Bytes/compare/master...MartyIX:jdk9-patch?expand=1#diff-77fac035fff19bf1e53777e3b28955d6R27 - I don't know how to wrap the class properly for it to work on released JDKs and JDK 9+

Tests pass for me. Hopefully, the patch will be of some use.

Regards,
Martin

### CODE REVIEW

**Comment by epickrram** [comment]
Hi @MartyIX , I've been working on the sun.misc.Cleaner/jdk.internal.ref.Cleaner problem, and now have a solution in place. I'll release chronicle-core tomorrow, which should take care of issue 2. The compiler arguments can also be placed in a property overridden in the profile - I'll dig out an example of that.
Once those are ready, if you'd be happy to update your patch, then I'll look at merging it later in the week.

**Comment by MartyIX** [comment]
@epickrram Thanks. I'll update the patch then :+1:

**Comment by epickrram** [comment]
@MartyIX I've addressed the two outstanding issues, can you merge master into your branch and see whether the tests are good?

**Comment by MartyIX** [comment]
Yes, I'll do it tomorrow morning when I get to my computer. 👍

File: pom.xml [hunk 1]
```diff
@@ -170,6 +170,7 @@
             <plugin>
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
+                <version>3.3.0</version>
```
Comment (MartyIX) [inline pom.xml]: This is in line with: https://github.com/OpenHFT/Chronicle-Core/blob/master/pom.xml#L147

**Comment by MartyIX** [comment]
I use the following XML in my pom.xml:

```

**Comment by MartyIX** [comment]
@epickrram I have updated my patch.

I use the following XML in my pom.xml:

```
        <dependency>
            <groupId>net.openhft</groupId>
            <artifactId>chronicle-core</artifactId>
            <scope>system</scope>
            <systemPath>${basedir}/../Chronicle-Core/target/chronicle-core-1.9.10-SNAPSHOT.jar</systemPath>
        </dependency>
```

When I run tests, they pass OK. Yet, there is the same warning as in `Chronicle-Core`:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running net.openhft.chronicle.bytes.ref.BinaryIntReferenceTest
[main] WARN net.openhft.chronicle.core.cleaner.CleanerServiceLocator - Error while trying to load service providers
java.util.ServiceConfigurationError: net.openhft.chronicle.core.cleaner.spi.ByteBufferCleanerService: Provider net.openhft.chronicle.core.cleaner.impl.jdk8.Jdk8ByteBufferCleanerService not found
	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:588)
	at java.base/java.util.ServiceLoader.access$200(ServiceLoader.java:390)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1206)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1215)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1259)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1294)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1379)
	at net.openhft.chronicle.core.cleaner.CleanerServiceLocator.cleanerService(CleanerServiceLocator.java:27)
	at net.openhft.chronicle.bytes.NativeBytesStore.<clinit>(NativeBytesStore.java:49)
	at net.openhft.chronicle.bytes.ref.BinaryIntReferenceTest.test(BinaryIntReferenceTest.java:32)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:272)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:236)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:386)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:323)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:143)
[main] WARN net.openhft.chronicle.core.cleaner.CleanerServiceLocator - Unable to find suitable cleaner service, falling back to using reflection
net.openhft.chronicle.core.Jvm: Unable to determine max direct memory
```

**Comment by epickrram** [comment]
You shouldn't need to specify chronicle-core-1.9.10-SNAPSHOT as a dependency. chronicle-bom-1.14.58 provides version 1.9.9 of core, which should contain everything you need. Can you try using the released version of chronicle-core, rather than one you've built locally?

There are two things we're trying to achieve here:

1. Java9 compatibility, meaning that the classes will compile/run on Java9+ (this involves fixing up the class changes such as sun.misc.Cleaner). This may involve some temporary kludges like using reflection.

2. Supporting Java9 runtimes, meaning that we take advantage of new features like the module system.

For point (2), there is a separate repository containing the runtime support: https://github.com/OpenHFT/Chronicle-JDK9

With this patch, I think we should just be trying to achieve (1), so we just need it to compile/run/pass using the released version of chronicle-core.

**Comment by MartyIX** [comment]
@epickrram I tried to run with and without chronicle-core-1.9.10-SNAPSHOT as local dependency and the `Jdk8ByteBufferCleanerService` warning appears only when I run tests with locally built chronicle-core-1.9.10-SNAPSHOT.

In both cases, tests were green.

**Comment by epickrram** [comment]
Awesome, thanks for your contribution.

**Pull request merged**